mono/mini/interp/interp.c

```diff
@@ -334,6 +334,33 @@ mono_interp_get_imethod (MonoDomain *domain, MonoMethod *method, MonoError *erro
 	return rtm;
 }
 
+#ifdef MONO_ARCH_HAS_MONO_CONTEXT
+#define INTERP_PUSH_LMF_WITH_CTX_BODY(ext, exit_addr) \
+	MONO_CONTEXT_GET_CURRENT ((ext).ctx); \
+	MONO_CONTEXT_SET_IP (&(ext).ctx, (exit_addr)); \
+	mono_arch_do_ip_adjustment (&(ext).ctx);
```

Comment:
@lewurm I'm assuming the only purpuse of ip adjustment is for EH to undo it, right ?

Reply:
correct